thrift.js

```diff
@@ -73,6 +87,25 @@ function Thrift(options) {
     self.link();
 }
 
+Thrift.loadSync = function loadSync(options) {
+    var filepath = path.resolve(options.thriftFile);
+
+    if (filepathThriftCache[filepath]) {
+        return filepathThriftCache[filepath];
+    }
+
+    options.source = fs.readFileSync(options.thriftFile, 'ascii');
+
+    var hash = crypto.createHash('md5')
+        .update(options.source)
+        .digest('hex');
+
+    var thrift = casThriftCache[hash] = filepathThriftCache[filepath] =
+        casThriftCache[hash] || new Thrift(options);
+
```

Comment:
This is where you need to drive separate compile and link steps. Don’t rely on the Thrift constructor to collect the transitive dependencies. Drive the transitive load+compile here (since compilation is necessary to infer deps) and then compile each thrift in the resultant "cache". Also, please call it a memo, not a cache.